app/utils/callback/lava.py

```diff
@@ -638,6 +481,177 @@ def store_rootfs_info(self, file_name):
             utils.LOG.warn("ValueError: {}".format(e))
 
 
+def store_artifacts(metadata, job_data, log):
+    store_test_log(metadata, log)
+    store_lava_json(metadata, job_data)
+
+
+def store_lava_json(metadata, job_data):
+    """ Save the json LAVA v2 callback object
+    Save LAVA v2 callback data as json file.
+
+    :param metadata: LAVA Callback metadata
+    :type metadata: dict
+    :param job_data: The JSON data from the callback.
+    :type job_data: dict
+    """
+
+    file_name = "-".join(["lava-json", metadata[models.DEVICE_TYPE_KEY]])
+    file_name = ".".join([file_name, "json"])
+
+    dir_path = metadata[models.DIRECTORY_PATH]
+
+    utils.LOG.info("Saving LAVA v2 callback file {} data in {}".format(
+        file_name,
+        dir_path))
+
+    file_path = os.path.join(dir_path, file_name)
+
+    # Removing the token
+    job_data.pop("token", None)
+
+    # Add extra information
+    job_data["lab_name"] = metadata.get("lab_name")
+    job_data["version"] = metadata.get("version")
+    job_data["boot_log_html"] = metadata.get("boot_log_html")
+    utils.make_path(dir_path)
+    with open(file_path, "wb") as f:
+        f.write(json.dumps(job_data))
+
+
+def store_test_log(metadata, log):
+    """Parse and save test logs
+
+    Parse the LAVA v2 log in YAML format and save it
+    as plain text and HTML.
+    :param metadata: LAVA Callback metadata
+    :type metadata: dict
+    :param log: LAVA log lines extracted from callback
+    :type log: list
+    """
+
+    dir_path = metadata[models.DIRECTORY_PATH]
+    suite = metadata[models.PLAN_KEY]
+    utils.LOG.info("Generating {} "
+                   "log files in {}".format(suite, dir_path))
+    file_name = "-".join([suite, metadata[models.DEVICE_TYPE_KEY]])
+    files = tuple(".".join([file_name, ext]) for ext in ["txt", "html"])
+    (metadata[models.BOOT_LOG_KEY],
+     metadata[models.BOOT_LOG_HTML_KEY]) = files
+    txt_path, html_path = (os.path.join(dir_path, f) for f in files)
+    utils.make_path(dir_path)
+    with codecs.open(txt_path, "w", "utf-8") as txt:
+        with codecs.open(html_path, "w", "utf-8") as html:
+            utils.lava_log_parser.run(log, metadata, txt, html)
+
+
+class LavaResults(object):
```

Comment:
Same thing here, adding a docstring may be good.

Reply:
added